app/controllers/content_item_signups_controller.rb

```diff
@@ -64,18 +69,15 @@ def assign_list_params
   end
 
   def params_service
-    if single_page_subscription?
+    if sign_up_to_content_id_based_subscription?
       SubscriberListParams::GenerateSinglePageListParamsService
     else
       SubscriberListParams::GenerateLinksBasedListParamsService
     end
   end
 
-  def single_page_subscription?
-    params[:single_page_subscription].present? && params[:single_page_subscription] == "true"
-  end
-
-  def render_confirm_page
-    render "confirm", locals: { single_page_subscription: single_page_subscription? }
+  def sign_up_to_content_id_based_subscription?
```

Comment:
Nitpick: That's a looooong method name... can we call it `content_id_subscription?` ?